openwisp_firmware_upgrader/base/models.py

```diff
@@ -612,6 +613,24 @@ def log_line(self, line, save=True):
         if save:
             self.save()
 
+        try:
+            # Convert lazy translations to strings to avoid serialization issues
+            line_str = str(line)  # Force evaluation of lazy translations
+            status_str = str(self.status)
+
+            # Publish to operation-specific channel
+            publisher = UpgradeProgressPublisher(self.pk)
+            publisher.publish_progress(
+                {"type": "log", "content": line_str, "status": status_str}
+            )
+
+            # Publish to device-specific channel for real-time UI updates
+            device_publisher = DeviceUpgradeProgressPublisher(self.device.pk, self.pk)
+            device_publisher.publish_log(line_str, status_str)
+
+        except Exception as e:
```

Comment:
expecting `Exception` is not a best practice. What could go wrong here? We should narrow down this to specific exception classes.